app/assets/javascripts/modules/improve-this-page.js

```diff
@@ -0,0 +1,164 @@
+(function (Modules) {
+  "use strict";
+
+  function ImproveThisPage () {
+    this.controller = null;
+    this.view = null;
+
+    this.start = function ($element) {
+      this.view = new View($element);
+      this.controller = new Controller(this.view);
+
+      this.controller.init();
+    }
+  };
+
+  function View ($element) {
+    var that = this;
+
+    this.$pageIsUsefulButton = $element.find('.js-page-is-useful');
+    this.$offerFeedbackButton = $element.find('.js-offer-feedback');
+    this.$feedbackFormContainer = $element.find('.js-feedback-form');
+    this.$feedbackForm = that.$feedbackFormContainer.find('form');
+    this.$feedbackFormSubmitButton = that.$feedbackFormContainer.find('[type=submit]');
+    this.$prompt = $element.find('.js-prompt');
+
+    this.onPageIsUsefulButtonClicked = function (callback) {
+      that.$pageIsUsefulButton.on('click', preventingDefault(callback));
+    }
+
+    this.onOfferFeedback = function (callback) {
+      that.$offerFeedbackButton.on('click', preventingDefault(callback));
+    }
+
+    this.onSubmitFeedbackForm = function (callback) {
+      that.$feedbackForm.on('submit', preventingDefault(callback));
+    }
+
+    this.replaceWithSuccess = function () {
+      $element.html('Thanks for your feedback.');
+    }
+
+    this.replaceWithGenericError = function () {
+      $element.html('Sorry, we’re unable to receive your message right now. We have other ways for you to provide feedback on the <a href=\"/contact/govuk\">contact page</a>.');
+    }
+
+    this.showFeedbackForm = function () {
+      that.$prompt.addClass('js-hidden');
+      that.$feedbackFormContainer.removeClass('js-hidden');
+    }
+
+    this.$feedbackFormContainerData = function () {
+      return that.$feedbackFormContainer.find('input, textarea').serialize();
+    }
+
+    this.$feedbackFormContainerTrackEventParams = function () {
+      return {
+        category: that.$feedbackFormContainer.data('track-category'),
+        action: that.$feedbackFormContainer.data('track-action')
+      }
+    }
+
+    this.pageIsUsefulTrackEventParams = function () {
+      return {
+        category: that.$pageIsUsefulButton.data('track-category'),
+        action: that.$pageIsUsefulButton.data('track-action')
+      }
+    }
+
+    this.renderErrors = function (errors) {
+      that.$feedbackFormContainer.find('.js-error').remove();
+
+      $.each(errors, function (attrib, messages) {
+        $.each(messages, function (index, message) {
+          var $errorNode = $('<div/>)', {
```

Comment:
I got it from an example in the jQuery docs. Search for "Create some DOM elements" on http://api.jquery.com/jquery/